File: backend/src/main/java/com/app/backend/dtos/UpdateUserDTO.java
package com.app.backend.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class UpdateUserDTO {
    @JsonProperty("ho_ten")
    private String hoTen;

    @JsonProperty("ten_hien_thi")
    private String tenHienThi;

    @JsonProperty("email")
    private String email;

    @JsonProperty("dia_chi")
    private String diaChi;

    @JsonProperty("password")
    private String password;

    @JsonProperty("retype_password")
    private String retypePassword;

}

File: frontend/src/app/dtos/nguoi-dung/update-user-dto.ts
export class UpdateUserDTO {
  ho_ten?: string;
  ten_hien_thi?: string;
  email?: string;
  dia_chi?: string;

  constructor(data?: Partial<UpdateUserDTO>) {
    if (data) {
      Object.assign(this, data);
    }
  }
}
